src/ota.rs

```diff
@@ -185,6 +185,19 @@ impl<'a> EspOtaUpdate<'a> {
     }
 }
 
+impl Drop for EspOtaUpdate<'_> {
```

Comment:
I think you need to call `core::mem::forget(self)` at the end of the `abort()` and `complete()` methods, or else you'll end up calling `esp_ota_abort` twice (once from `abort` and once more from the dtor). Or you'll call `esp_ota_abort` post-completion, which should not happen either, if the user happens to invoke `complete` instead.

See [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f72c0b1023f12f0978f812888731f916).

Reply:
Ah, d'oh, of course.  Though, one downside of using `mem::forget` is that it's an easy thing to forget or accidentally remove.

Another option would be to instead make the `update_handle` field `Option<esp_ota_handle_t>`.  All methods that use it can `.take().unwrap()` it since we would never construct an instance without it set, but then the drop implementation can use `if let Some(handle) = self.update_handle.take()` to conditionally run it only if it was not already taken.  What do you think about that?

Reply:
I still think putting `forget` at two places is perhaps the simpler approach. Also what I found online as the recommended approach. :)

Reply:
Fair enough.  :+1:

Though if we define the behavior to be abort on drop then I guess the abort method can just be empty, no need to duplicate the abort implementation.

Reply:
Also an option, yes. Or just pass on the `abort` method completely, but that might mean rethinking the traits too, and given that a release is imminent, perhaps we can pass on that for now.